### ISSUE TITLE
FIne-tuning

### ISSUE THREAD

**Issue by vinnytwice** [cdqa-suite/cdQA#280]
Hi, Thank you very much for this library.
I'm trying it on your notebooks and I have a doubt.

When you fit the model with `dqa_pipeline.fit_reader(path_to_data)` do you retrain the entire model or just the last layer? 
I mean, do you keep all the weights from the pretrained model?
Sorry for the very noob question but I just started with ML and python..
Thank you again.

**Comment by andrelmfarias** [comment]
Hi @vinnytwice,

It is a good question. Actually during the fine-tune the whole model is retrained, but the weights are initialized for BERT pre-trained weights.

**Comment by vinnytwice** [comment]
Hi Andrel thank you for answering
I'm not sure I follow you, as I'm very new to all this ML thing..
As I understood by checking other repos and posts, If you use a pre trained model you use it's weights and if you train your model you just build new weights.
Others freeze all the hidden layers except the last and fine-tune ( aka re-train) only that layer..
so cdqa is not fine-tuneable, but only trainable right?
Did I get it wrong?

**Comment by andrelmfarias** [comment]
No worries @vinnytwice ,

Actually, when you fine-tune a model there are two approaches:
1. Freeze all hidden layers except the last-one and train for a **few** epochs.
2. Start from the pre-trained model and train it for a **few** epochs on your data. Sometimes you can even fine-tune the hidden layers with a smaller learning rate than the one you use to train the last layer.

In cdQA we followed the same approach as HuggingFace and Google AI team to fine-tune the model on SQuAD: fine-tune the whole model for a few epochs (2 in the case of the models we make available) with the same learning-rate.

It's also considered a fine-tune because you start from the pre-trained model and train it for only a few epochs.
